command/agent/command.go

```diff
@@ -277,10 +269,28 @@ func (c *Command) readConfig() *Config {
 		return config
```

Comment:
Move the `if dev` check into the new function. This will allow us to still validate things that are required even in dev mode (such as the meta variables)

Reply:
I would organize the validate function as follows: validation that applies in both dev and non, the if dev check that returns early, and then all the non-dev mode checks

Reply:
There currently isn't any stand-alone validation on `dev` ("before the `if dev`"); what little existing dev-mode validation is inline with config discovery; e.g., normalizeAddrs():
https://github.com/hashicorp/nomad/blob/f-1157-validate-node-meta-variables/command/agent/config.go#L869

Are you suggesting that I go through the existing validation and see which **should** apply in dev and non-dev? This would be a change of behavior.